08-rails/test-02-setup.md

```diff
@@ -0,0 +1,97 @@
+
+Then run `bundle install`
+
+#### Step 2:  Installing Minitest Spec
+
+Then we will ask Rails to modify test_helper.rb to use spec style describe and it blocks with `rails generate minitest:install`  It will ask you if you want to overwrite the `test_helper.rb` file.  Answer yes **(Y)**.
+
+#### Step 3:  Make Spec-style Testing the default
+
+In `config/application.rb` add these lines to the `Application` class.
+
+```ruby
+config.generators do |g|
+  g.test_framework :minitest, spec: true
+end
+```
+
+#### Step 4:  Use Minitest Reporters (Optional)
+
+Lastly edit test/test_helper.rb to tell Rails to use the Minitest-spec style tests with `atom test/test_helper.rb` and add these lines to the top.
+
+```ruby
+require "minitest/reporters"  # for Colorized output
+
+#  For colorful output!
+Minitest::Reporters.use!(
+  Minitest::Reporters::SpecReporter.new,
+  ENV,
+  Minitest.backtrace_filter
+)
+```
+
+## Making Spec-style Testing the Default
+
+As of right now, you'll have to follow all the above steps for every new rails project you make. Sounds like an easy way to forget something. Fortunately, Rails allows you to make all this stuff happen by default, every time you say `rails new`.
+
+Remember the `~/.railsrc` file? Previously we added a line to make postgres the default database for all new Rails projects.
+
+We're going to do something similar now. The big difference is that making Rails use spec-style testing is a lot more complex than just changing the default database. We'll use a combination of the `.railsrc` file and a new piece of rails called a _template_.
+
+What is a _template_? For now, just think of it as a way to add some more complex defaults to Rails. Rails templates are a rabbit hole we could spend a whole bunch of time exploring, but instead we're just going to supply one for you.
+
+### Instructions
+
+1. Download [this template file](https://gist.github.com/droberts-ada/26a3bff1510b0b45fa5dd764a833483c), and save it as `~/.rails-template.rb`
+1. Run the following command in your terminal
+    ```bash
+    echo "-m ~/.rails-template.rb" >> ~/.railsrc
+    ```
+
+That's it, you should now be good to go. From now on, every time you type `rails new` Rails will set your project up for spec-style testing. You shouldn't need to repeat these instructions until the next time you buy a computer.
+
+### Trust But Verify
+
+To verify things worked as expected, create a new rails project. You should see a few things:
+
+- At the end of the output from `rails new`, you should see something like
+    ```
+    generate  minitest:install
+    Running via Spring preloader in process 51275
+    force  test/test_helper.rb
+    insert  test/test_helper.rb
+    ```
+- Our two testing gems, `minitest-rails` and `minitest-reporters`, should be included in the `Gemfile`
+- Code matching what's above should have been inserted into `config/application.rb` and `test/test_helper.rb`
```

Comment:
Do you have a link that helped you create the template structure? It might be nice to just add an "Additional Resources" if folks are interested in exploring the documentation on this more